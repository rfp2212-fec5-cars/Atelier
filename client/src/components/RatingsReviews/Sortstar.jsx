import React from 'react';


var Sortstar = ({ k, rateStar, handleUserClick }) => {
  //console.log(k);
  //console.log(rateStar);
  const handleSortStar = ()=>{
    handleUserClick(k);
  };
  return (
    <div className='sortstar' onClick={handleSortStar}>
      <span>{k} Stars </span>
      <div className="greybar">
        <div className='greenbar' style={{ height: '1em', width: `${rateStar[k] + '%'}` }}></div>
      </div>
      <span>{rateStar[k]}</span>
    </div>
  );

};
export default Sortstar;