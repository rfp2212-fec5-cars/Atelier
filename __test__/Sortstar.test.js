import TestRenderer from 'react-test-renderer';
import Sortstar from '../client/src/components/RatingsReviews/Sortstar.jsx';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render, screen, fireEvent } from '@testing-library/react';



describe('ProductCharacteristic testing', () => {
  it('should exists', () => {
    const testRenderer = TestRenderer.create(
      <Sortstar k={5} rateStar={[-1, '11.94', '5.97', '23.88', '6.72', '51.49']}></Sortstar>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have one span to distinguish stars', () => {
    const testRenderer = TestRenderer.create(
      <Sortstar k={5} rateStar={[-1, '11.94', '5.97', '23.88', '6.72', '51.49']}></Sortstar>
    );
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node => node.type === 'span')).toHaveLength(1);
  });

  test('should invoke handleUserClick once user click', async () => {
    const handleUserClick = jest.fn();
    const utils = render(<Sortstar k={5} rateStar={[-1, '11.94', '5.97', '23.88', '6.72', '51.49']} handleUserClick={handleUserClick}></Sortstar>)
    const sort = utils.getByTestId('sortstar')
    expect(sort).toBeTruthy();


    fireEvent.click(sort)
    expect(handleUserClick).toHaveBeenCalledTimes(1);

  });

});


