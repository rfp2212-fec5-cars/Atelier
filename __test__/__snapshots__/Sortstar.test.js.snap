// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCharacteristic testing should exists 1`] = `
<div
  className="sortstar"
  data-testid="sortstar"
  onClick={[Function]}
>
  <span>
    5
     Stars 
  </span>
  <div
    className="greybar"
  >
    <div
      className="greenbar"
      style={
        {
          "height": "1em",
          "width": "51.49%",
        }
      }
    />
  </div>
</div>
`;

exports[`sortStar testing should exists 1`] = `
<div
  className="sortstar"
  data-testid="sortstar"
  onClick={[Function]}
>
  <span>
    5
     Stars 
  </span>
  <div
    className="greybar"
  >
    <div
      className="greenbar"
      style={
        {
          "height": "1em",
          "width": "51.49%",
        }
      }
    />
  </div>
</div>
`;
